rotkehlchen/chain/evm/decoding/uniswap_v2_like/decoder.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Literal
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.modules.uniswap.v2.constants import SWAP_SIGNATURE
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import ActionItem
+from rotkehlchen.chain.evm.decoding.uniswap_v2_like.common import (
+    compute_uniswap_v2_like_pool_address_with_unpacked_addresses,
+    decode_uniswap_like_deposit_and_withdrawals,
+    decode_uniswap_v2_like_swap,
+)
+from rotkehlchen.chain.evm.decoding.uniswap_v2_like.constants import (
+    BURN_SIGNATURES,
+    MINT_SIGNATURE,
+)
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class UniswapV2LikeDecoder(DecoderInterface):
+    weth_asset: EvmToken | None = None
```

Comment:
why make this a part of the class and initialize to None? This should be set by the constructor and should always have a value. That's not how inheritance works.